plugins/hydra_ray_launcher/hydra_plugins/hydra_ray_launcher/ray_launcher.py

```diff
@@ -0,0 +1,211 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+import logging
+import os
+import tempfile
+from enum import Enum
+from pathlib import Path
+from subprocess import PIPE, Popen
+from typing import Any, Dict, List, Sequence, Tuple
+
+import ray.cloudpickle as cloudpickle  # type: ignore
+from omegaconf import DictConfig, OmegaConf, open_dict
+
+from hydra.core.config_loader import ConfigLoader
+from hydra.core.config_search_path import ConfigSearchPath
+from hydra.core.hydra_config import HydraConfig
+from hydra.core.singleton import Singleton
+from hydra.core.utils import JobReturn, configure_log, filter_overrides, setup_globals
+from hydra.plugins.launcher import Launcher
+from hydra.plugins.search_path_plugin import SearchPathPlugin
+from hydra.types import TaskFunction
+
+log = logging.getLogger(__name__)
+
+
+class RayLauncherMode(Enum):
+    LOCAL = "local"
+    AWS = "aws"
+
+
+class RayLauncherSearchPathPlugin(SearchPathPlugin):
+    def manipulate_search_path(self, search_path: ConfigSearchPath) -> None:
+        # Appends the search path for this plugin to the end of the search path
+        search_path.append(
+            "hydra-ray-launcher", "pkg://hydra_plugins.hydra_ray_launcher.conf"
+        )
+
+
+class RayLauncher(Launcher):
+    def __init__(self, **params: DictConfig) -> None:
+        self.config = None
+        self.task_function = None
+        self.config_loader = None
+        self.mode = RayLauncherMode(params["mode"])
```

Comment:
keep this in mind when you switch to Structured configs.
You instincts are already sending you toward more type safety.